src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/stateful_transpose_sdpa_fusion.cpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "stateful_transpose_sdpa_fusion.hpp"
+
+#include <cstdint>
+#include <limits>
+#include <openvino/core/rt_info.hpp>
+#include <openvino/opsets/opset1.hpp>
+#include <openvino/opsets/opset13.hpp>
+#include <openvino/opsets/opset6.hpp>
+#include <openvino/opsets/opset8.hpp>
+#include <openvino/pass/pattern/op/or.hpp>
+#include <openvino/pass/pattern/op/wrap_type.hpp>
+#include <transformations/utils/utils.hpp>
+
+#include "itt.hpp"
+#include "ov_ops/type_relaxed.hpp"
+#include "transformations/cpu_opset/common/op/sdpa.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+StatefulTransposeSDPAFusion::StatefulTransposeSDPAFusion() {
+    MATCHER_SCOPE(StatefulTransposeSDPAFusion);
```

Comment:
highly recommend merge this `StatefulTransposeSDPAFusion` with `StatefulMultiQuerySDPAFusion`, since `StatefulMultiQuerySDPAFusion` has very small sub-pattern related to `multi-query` which can be made to be optional